gen/generate_test.go

```diff
@@ -62,3 +62,51 @@ func TestGenerateWithRelativePaths(t *testing.T) {
 		}
 	}
 }
+
+func TestThriftPackageImporter(t *testing.T) {
+	importer := thriftPackageImporter{
+		ImportPrefix: "github.com/myteam/myservice",
+		ThriftRoot:   "/src/thrift",
+	}
+
+	tests := []struct {
+		File, ServiceName, Relative, Package, ServicePackage string
+	}{
+		{
+			File:           "/src/thrift/foo.thrift",
+			Relative:       "foo",
+			Package:        "github.com/myteam/myservice/foo",
+			ServiceName:    "MyService",
+			ServicePackage: "github.com/myteam/myservice/foo/myservice",
+		},
+		{
+			File:     "/src/thrift/shared/common.thrift",
+			Relative: "shared/common",
+			Package:  "github.com/myteam/myservice/shared/common",
+		},
+	}
+
+	for _, tt := range tests {
+		if tt.Relative != "" {
```

Comment:
Yeah, for all existing test cases, but I'm treating it as optional. If Relative is provided, that's the expected `RelativePackage`, if `ServicePackage` is provided, that`s the expected `ServicePackage` and so on.